core/src/main/scala/kafka/utils/KafkaScheduler.scala

```diff
@@ -120,7 +120,8 @@ class KafkaScheduler(val threads: Int,
             trace("Completed execution of scheduled task '%s'.".format(name))
           }
         }
-        if (period >= 0)
+        //fixbug: if period==0 , it should be an execution thread rather than a throw error
+        if (period > 0)
```

Comment:
Nice catch! Please remove the comment, and add a test in `SchedulerTest` for it. Thanks.

Reply:
Ok, thank you

Reply:
> Nice catch! Please remove the comment, and add a test in `SchedulerTest` for it. Thanks.

It has been completed